### PR TITLE
Discount coupon fix

### PR DESCRIPTION
**Pull request by mayankamencherla**
Fixed amount mismatch when discount coupon is applied

### CODE REVIEW

File: razorpay-payments.php [hunk 1]
```diff
@@ -222,7 +222,7 @@ protected function verifyOrderAmount($razorpayOrderId, $orderId)
 
             $razorpayOrderArgs = array(
                 'id'        => $razorpayOrderId,
-                'amount'    => (int) $order->order_total*100,
+                'amount'    => $order->order_total*100,
```
Comment (captn3m0) [inline razorpay-payments.php]: why was this changed?

Comment (mayankamencherla) [inline razorpay-payments.php]: @captn3m0 It was rounding 199 with 20% discount to 15919 instead of 15920

Comment (captn3m0) [inline razorpay-payments.php]: Dropping the int also means we may be passing float to our systems lower down which may cause issues, no? Did some tests:

```>>> $a  = 159.2;
=> 159.2
>>> (int) $a * 100;
=> 15900
>>> (int) ($a * 100);
=> 15919
>>> 159.2 * 100
=> 15920.0
>>> (int)(159.2 * 100);
=> 15919
>>> (159.2 * 100);
=> 15920.0
>>> floor($a * 100);
=> 15919.0
>>> round($a * 100);
=> 15920.0
>>> round($a * 100);
=> 15920.0
>>> intval($a * 100);
=> 15919
>>> ceil($a * 100);
=> 15920.0
>>> round($a * 100);
=> 15920.0
>>> (int) round ($a*100);
=> 15920
```

The typecast + round approach looks cleaner to me. /cc @73sl4

Comment (mayankamencherla) [inline razorpay-payments.php]: @captn3m0 Cool. Agreed. Pushed the code.

**Comment by captn3m0** [comment]
Please test locally and do a squash merge.
